File: 250126/정수 명령 처리 6/process-numeric-commands-6.cpp
#include <iostream>
#include <queue>
using namespace std;

int n,num;
string op;
priority_queue<int> pq;
int main() {
    
    cin >> n;
    for(int i=0; i<n; i++){
        cin >> op;
        if(op == "push"){
            cin >> num;
            pq.push(num);
        }  
        else if(op == "size") cout << pq.size() << '\n';
        else if(op == "empty") cout << pq.empty() << '\n';
        else if(op == "pop") {
            cout << pq.top() << '\n';
            pq.pop();
        }
        else cout << pq.top() << '\n';
    }
    return 0;
}

File: 250319/정수 명령 처리 6/process-numeric-commands-6.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        int n = sc.nextInt();
        for(int i=0; i<n; i++){
            String op = sc.next();
            if(op.equals("push")){
                int num = sc.nextInt();
                pq.add(-num);
            }
            else if(op.equals("size")) System.out.println(pq.size());
            else if(op.equals("empty")){
                if(pq.isEmpty()) System.out.println(1);
                else System.out.println(0);
            }
            else if(op.equals("pop")) System.out.println(-pq.poll());
            else System.out.println(-pq.peek());
        }
        sc.close();
    }
}